player/client.c

```diff
@@ -1586,6 +1586,37 @@ int mpv_unobserve_property(mpv_handle *ctx, uint64_t userdata)
     return count;
 }
 
+static bool property_shared_prefix(const char *a0, const char *b0)
+{
+    bstr a = bstr0(a0);
+    bstr b = bstr0(b0);
+
+    // Treat options and properties as equivalent.
+    bstr_eatstart0(&a, "options/");
+    bstr_eatstart0(&b, "options/");
+
+    // Make sure a is >= b's length
+    if (a.len < b.len) {
+        bstr tmp = a;
+        a = b;
+        b = tmp;
+    }
```

Comment:
might be worth shortcutting the common case here where `a.len == b.len` (that's common, right?)

Reply:
Not sure what you mean?

Reply:
```c
} else if (a.len == b.len) {
    return bstrcmp(a, b) == 0;
}
```

Reply:
I've rewritten this function to be a bit simpler.